src/main.rs

```diff
@@ -56,7 +76,7 @@ fn collapse_bits(bytes : &[u8]) -> Result<u8, String> {
             0 => result |= mask,
             1 => result &= !mask,
             _ => {
-                return Err(format!("Bits must be either 0 or 1 (value was {})", *byte));
+                return Err(FontCreationError::new(format!("Bits must be either 0 or 1 (value was {})", *byte)));
```

Comment:
```rust
bail!("Bits must be either 0 or 1 (value was {})", *byte);
```

Reply:
Looks like both this and `ensure!` are unhappy about the error type:

```
  --> src/main.rs:79:17
   |
79 |                 bail!("Bits must be either 0 or 1 (value was {})", *byte);
   |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected struct `sss_fontbuild::errors::FontCreationError`, found struct `quicli::prelude::Error`
   |
   = note: expected type `sss_fontbuild::errors::FontCreationError`
              found type `quicli::prelude::Error`
   = note: this error originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)
```

Is the `failure` create necessary for this to work?

Reply:
Yeah, the macro can't know how to construct your error with a string otherwise. (It calls failure's `err_msg` [internally](https://docs.rs/failure/0.1.1/src/failure/macros.rs.html#10-17) which returns a `failure::Error`.)

If you wanted, you could probably copypaste these macros into your code quite easily, though.

Reply:
Actually, I haven't tested if this works with quicli's re-export! You might need to include failure directly.

Reply:
> Actually, I haven't tested if this works with quicli's re-export!

With some quick testing, it looks like that's the case. quicli's re-export doesn't *appear* to make `Fail` available. For example, trying to use it nets this error:

```
pub struct FontCreationError {
    message: String,
}
```

Actually, if I end up directly importing the `failure` crate via 

```
extern crate failure;
#[macro_use] extern crate failure_derive;
```

It seems to also conflict with quicli's reexport; I receive a warning about `conflicting implementation in crate 'failure'` unless I remove the quicli imports from the project root.

Reply:
Please ignore my earlier edit about the conflict - I misread an error message. However, it is the case that it doesn't seem to be possible to use `Fail` without independently importing both `failure` and `failure_derive`.

Reply:
The code failure_derive 0.1.1 produces seems to insert `extern crate failure;` but I can't find the code as the crate seems to be undergoing quite a large refactoring. As probably _all_ the comments here already tell you: We need to have a better story integrating failure :)